### PR TITLE
Fix goroutine leak

### PR DESCRIPTION
**Pull request by evrardjp**
Without this patch, we use WriterLevel, which spawns go routines. As we do it at every call of the util commands, we spawn goroutines at every check.

This is a problem as it leads to memory management issues.

This fixes it by using a buffer for stdout and stderr, then logging the results after the command was executed.

To make sure the logging happened at the same place, I inlined the code from utils. This results in duplicated the code.

However, this is not a big problem as:
- It makes the code more readable
- The implementation between checkers and rebooters _ARE_ different -- One definitely NEEDS privileges, while the other does not... Which could lead to later improvements.

Removing a "utils" package is not really a big deal (it is kinda a win in itself, as it is an anti-pattern), as the test coverage was kept.

Partial-Fix: #1004
Fixes: #1013

### CODE REVIEW

**Comment by evrardjp** [comment]
@jonaz Could you check this, please?

File: pkg/checkers/checker.go [hunk 1]
```diff
@@ -53,16 +53,15 @@ type CommandChecker struct {
...
-	cmd := util.NewCommand(cmdline[0], cmdline[1:]...)
+	bufStdout := new(bytes.Buffer)
+	bufStderr := new(bytes.Buffer)
+	cmd := exec.Command(rc.CheckCommand[0], rc.CheckCommand[1:]...)
+	cmd.Stdout = bufStdout
+	cmd.Stderr = bufStderr
+
 	if err := cmd.Run(); err != nil {
 		switch err := err.(type) {
 		case *exec.ExitError:
```
Comment (jonaz) [inline pkg/checkers/checker.go:67]: We might want to output stderr and stdout in the log.Warnf below so the user gets important output. Previously this was displayed with the io.Pipe in logrus writer.

File: pkg/reboot/command.go [hunk 2]
```diff
@@ -15,9 +16,17 @@ type CommandRebooter struct {
...
-	if err := util.NewCommand(c.RebootCommand[0], c.RebootCommand[1:]...).Run(); err != nil {
+
+	bufStdout := new(bytes.Buffer)
+	bufStderr := new(bytes.Buffer)
+	cmd := exec.Command(c.RebootCommand[0], c.RebootCommand[1:]...)
+	cmd.Stdout = bufStdout
+	cmd.Stderr = bufStderr
+
+	if err := cmd.Run(); err != nil {
 		return fmt.Errorf("error invoking reboot command %s: %v", c.RebootCommand, err)
```
Comment (jonaz) [inline pkg/reboot/command.go:27]: We do want to add both stderr and stdout to the error returned. Could contain important info about the error.

Comment (evrardjp) [reply pkg/reboot/command.go:27]: sure.

File: pkg/checkers/checker.go (hunk 1 above)
Comment (evrardjp) [reply pkg/checkers/checker.go:67]: fine for me.

Comment (evrardjp) [reply pkg/checkers/checker.go:67]: Side note, when error happens, most likely the stderr and stdout will be empty. Do we really need this?

**Comment by evrardjp** [comment]
@jonaz  can you have a second look please?

File: pkg/checkers/checker.go (hunk 1 above)
Comment (jonaz) [reply pkg/checkers/checker.go:68]: it totally depends on the command and hiding the error is not helpful for the end user.

**Comment by evrardjp** [comment]
I tested commit 1 on a new env. It has good results.

**Comment by jonaz** [comment]
@evrardjp LGTM :+1:

**Review by jonaz — APPROVED** [review]

**Review by dholbach — APPROVED** [review]

**Pull request merged**